include/swift/AST/DiagnosticsSema.def

```diff
@@ -394,6 +394,46 @@ ERROR(cannot_convert_closure_result_protocol,none,
 ERROR(cannot_convert_closure_result_nil,none,
       "'nil' is not compatible with closure result type %0", (Type))
 
+NOTE(generic_argument_mismatch,none,
+     "arguments to generic parameter %0 (%1 and %2) are expected to be equal",
+     (Identifier, Type, Type))
+ERROR(cannot_convert_generic_type_argument,none,
```

Comment:
Some of the diagnostics like this one already exist in here you can grep for "cannot convert value of" to figure out what their names are.